src/Microsoft.NetCore.Analyzers/Core/Runtime/InstantiateArgumentExceptionsCorrectly.Fixer.cs

```diff
@@ -13,17 +21,125 @@ public abstract class InstantiateArgumentExceptionsCorrectlyFixer : CodeFixProvi
     {
         public sealed override ImmutableArray<string> FixableDiagnosticIds => ImmutableArray.Create(InstantiateArgumentExceptionsCorrectlyAnalyzer.RuleId);
 
-        public sealed override FixAllProvider GetFixAllProvider()
+        public sealed override FixAllProvider GetFixAllProvider() => WellKnownFixAllProviders.BatchFixer;
+
+        public sealed override async Task RegisterCodeFixesAsync(CodeFixContext context)
         {
-            // See https://github.com/dotnet/roslyn/blob/master/docs/analyzers/FixAllProvider.md for more information on Fix All Providers
-            return WellKnownFixAllProviders.BatchFixer;
+            var document = context.Document;
+            var cancellationToken = context.CancellationToken;
+            var root = await document.GetSyntaxRootAsync(cancellationToken).ConfigureAwait(false);
+            var semanticModel = await document.GetSemanticModelAsync(cancellationToken).ConfigureAwait(false);
+
+            var objectCreation = root.FindNode(context.Span, getInnermostNodeForTie: true);
+            var diagnostic = context.Diagnostics.First();
+
+            switch (diagnostic.Properties["Problem"])
+            {
+                case "NoArguments":
+                    await RegisterNoArgumentsCodeFixesAsync().ConfigureAwait(false);
+                    break;
+                case "IncorrectMessage":
+                    break;
+                case "IncorrectParameterName":
```

Comment:
@CyrusNajmabadi I want to implement something where if a parameter name is slightly misspelled, the fixer detects that and offers a fix for that parameter. e.g. `new ArgumentOutOfRangeException("idnex")` => `new ArgumentOutOfRangeException(nameof(index))`. Is it possible to use the word distance algorithms in Roslyn from this repo?

Reply:
This seems like basically the same thing as CA1507 (or at least conceptually, something that CA1507 should do). Whatever we do here, we should also make CA1507 do.

Reply:
As to `WordSimilarityChecker`, it's currently and internal API, so we'd have to make it public. That's a rather big process (certainly nothing that will happen in the 15.6 timeframe). @jinujoseph, do we think that API is something we'd consider making publicly-available?

Reply:
Async002 also seems like it could benefit from this... if a method doesn't end in `Async` because `Async` is misspelled, like `FooAsnyc`, we want to detect that and offer `FooAsync` instead of `FooAsnycAsync`.